src/Components/homepage/Homepage.js

```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
 import Card from '../Card/Card'
 import { fetchData } from '../../Apis/Dummy-api'
 import '../Style/Homepage.css'
+import RecentlyPlayedZone from '../Recently-played-zone/Recently-played-zone'
 
 export const Homepage = () =>  {
     const [cards, setCards] = useState([])
 	const [errorMessage, setErrorMessage] = useState([])
-
+	const [recentlyPlayedCards, setRecentlyPlayedCards] = useState([])
 	const numberOfCards = 20
 
     useEffect(() => {
-        fetchData(numberOfCards).then((res) => {setCards(res)}).catch((e) => {
-            setErrorMessage(e)
+        fetchData(numberOfCards)
+        .then((res) => {setCards(res)})
+        .catch((error) => {
+            setErrorMessage(error)
             console.error('[ERROR IN HOMEPAGE COMPONENT FETCH DATA]: ' + errorMessage)
         })
     }, [])
 
+    // useEffect(() => {
+    //     if (cards.length) {
+    //         console.log(cards)
+    //         const firstCards = cards.slice(0,4)
+    //         setRecentlyPlayedCards(firstCards)
+    //     }
+    // }, [cards])
+
+    const addRecentCard = (clickedCardId) => {
+        setRecentlyPlayedCards((oldCardsArray) => {
+            const found = oldCardsArray.find((element) => element === cards[clickedCardId - 1])
+            if (found) {
+                return [...oldCardsArray]
+            }
```

Comment:
why do we return the old elements if the clicked one is found? 🤔 
i think you can just invert the condition and omit this step